### ISSUE TITLE
TextInput: multiline rich content support

### ISSUE THREAD

**Issue by necolas** [necolas/react-native-web#1023]
Multiline `TextInput` in React Native supports components as children. Along with #795 and #793, this may require switching to `contentEditable` for the `multiline` implementation, if not the single-line input.

**Comment by steida** [comment]
Check https://github.com/ianstormtaylor/slate, it seems to be the best contentEditable lib.

**Comment by voidstarfire** [comment]
Slate doesn't play so nice with mobile unfortunately. Same with DraftJS.

The easiest solution is just wrap each one in it's own native renderer function that spits out the following:

SubComponent
```
<Text>
     <Text style={{ fontWeight: 'bold' }}>Starts </Text>bold.
</Text>
```

Web
```
<p contentEditable={true} onKeyDown={(e) => this.onChange(e)}>
    <SubComponent />
</p>
```

Mobile
<TextInput 
    multiline={true} 
    onChange={(e) => this.onChange(e)}
>
    <SubComponent />
</TextInput>

PS Really awesome library I'm using this to create cross platform apps (Web, Electron, React-Native) and it's going epic. Wish I could donate!

**Comment by necolas** [comment]
@voidstarfire Thanks! How does React manage when a user edits the contentEditable content that is produced by components React renders? I heard that was one of the difficulties with contentEditable but haven't made time to experiment myself yet. Any more feedback you can include would be really helpful

**Comment by voidstarfire** [comment]
I'm playing with that just now. 

It seems to work really well so far. The ```<Text/>``` components just wrap around the text.

Seems we have to create some sort of generator to filter out empty text components and fill in styled ones. Should work well with React-native-web. 

I will get back probably mid next week with my results :-)

**Comment by necolas** [comment]
> Seems we have to create some sort of generator to filter out empty text components and fill in styled ones.

What does that mean?

**Comment by shakeabi** [comment]
@ollydixon How is the development going?

**Comment by ollydixon** [comment]
@shakeabi I'm using a library called Quill, worked great for mobile and web :-)

**Comment by shakeabi** [comment]
@ollydixon How are you using quilljs in native cuz there's no native supported quill yet, right? (If webview, then though it may work for various devices, it's still not purely native and can have unexpected behaviour)

**Comment by ollydixon** [comment]
@shakeabi there was no native solution that included an editor.

**Comment by shakeabi** [comment]
@ollydixon It's not that difficult to make a simple native rich text editor. My (partial) solution includes using nested <Text> inside <TextInput>. However, react-native-web doesn't support this as it renders either textarea or input tags (neither of them supports children tags). #1380: I've discussed the same in this issue. A probable solution would be to replace the TextInput version (textarea or input) react-native-web to content-editable divs.

**Comment by ollydixon** [comment]
@shakeabi yes I discovered this as well. I no longer do React Native anymore though. Flutter is much better :-)
